lua/snips.lua

```diff
@@ -64,20 +64,26 @@ function M:yank_shared_url(cleaned_output, silent)
 end
 
 ---Execute the ssh command line to send the file on snips
-function M.execute_snips_command()
-    local selected_lines = {}
-    for _, line in ipairs(M.get_selected_lines()) do
-        table.insert(selected_lines, line)
-    end
+function M.execute_snips_command(args)
+    local content
+    local ext
+    if not (args and args.from_register) then
+        local selected_lines = {}
+        for _, line in ipairs(M.get_selected_lines()) do
+            table.insert(selected_lines, line)
+        end
 
-    if selected_lines == nil or #selected_lines == 0 then
-        vim.print("SNIPS::ERROR:: No lines selected.")
-        return
-    end
+        if selected_lines == nil or #selected_lines == 0 then
+            vim.print("SNIPS::ERROR:: No lines selected.")
+            return
+        end
 
-    local content = table.concat(selected_lines, "\n")
+        content = table.concat(selected_lines, "\n")
+        ext = M.get_file_extension()
```

Comment:
interesting point, if it's coming from the clipboard we should let snips.sh infer the extension from the code provided 👌🏽